### PR TITLE
Stopped unset properties from being serialized and added in conditional property serialization for collections in JsonWebKey

### LINKED ISSUE #579: Do not serialize some properties on JsonWebKey
When trying to serialize an instance of JsonWebKey there are a few properties that are showing up on the wire even though they are not set.

```
{
  "key_ops": [],
  "kid": "DevelopmentKey.5efea89c-6b7f-4ab3-9579-14b1134a4636",
  "kty": "RSA",
  "e": "AQAB",
  "n": "",
  "use": "sig",
  "x5c": [],
  "keySize": 2048,
  "hasPrivateKey": false,
  "keyId": null,
  "cryptoProviderFactory": {
      "customCryptoProvider": null
}
```

It would be good if you could hide these properties by using JsonIgnore http://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonIgnoreAttribute.htm

(KeySize, HasPrivateKey, KeyId, CryptoProviderFactory)

It would also be good if you can add a setter to the collection properties so that they can be set to null and not send over the wire when empty.

This issue is related to https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/578

### PR DESCRIPTION
**Pull request by mafurman**
Fixes #579.

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs [hunk 1]
```diff
@@ -280,6 +283,24 @@ public bool HasPrivateKey
             }
         }
 
+        /// <summary>
+        /// Gets a bool that determines if the 'key_ops' (Key Operations) property should be serialized.
+        /// </summary>
+        /// <return>true if 'key_ops' (Key Operations) is not empty; otherwise, false.</return>
+        public bool ShouldSerializeKeyOps()
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: Where is this used?

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs [hunk 2]
```diff
@@ -280,6 +283,24 @@ public bool HasPrivateKey
...
+        {
+            return KeyOps.Any();
+        }
+
+        /// <summary>
+        /// Gets a bool that determines if the 'x5c' collection (X.509 Certificate Chain) property should be serialized.
+        ///</summary>
+        /// <return>true if 'x5c' collection (X.509 Certificate Chain) is not empty; otherwise, false.</return>
+        public bool ShouldSerializeX5c()
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: Where is this used?

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs [hunk 3]
```diff
@@ -280,6 +283,24 @@ public bool HasPrivateKey
...
+        {
+            return X5c.Any();
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: Don't need Linq for this. X5c is an IList. If method is needed, just use X5c.Count > 0

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs (hunk 1 above)
Comment (mafurman) [reply src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: I believe this is called automatically when serialization occurs. I just followed the instructions here:
https://www.newtonsoft.com/json/help/html/ConditionalProperties.htm

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs (hunk 2 above)
Comment (mafurman) [reply src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: See the comment above.

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs (hunk 3 above)
Comment (mafurman) [reply src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: Just made this change for both methods.

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs (hunk 2 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: oh, so does this puts the serialization under user control?

File: src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs (hunk 1 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: Can you add a to the comments that JSON.Net uses this.

Comment (mafurman) [reply src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs]: Just added this in for both methods.

**Comment by brentschmaltz** [comment]
:shipit:

**Pull request closed without merging**